core/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -618,16 +618,16 @@ private static void parseNullValue(ParseContext context, ObjectMapper parentMapp
 
     private static Mapper.Builder<?,?> createBuilderFromFieldType(final ParseContext context, MappedFieldType fieldType, String currentFieldName) {
         Mapper.Builder builder = null;
-        if (fieldType instanceof StringFieldType) {
```

Comment:
Moving my original comment from #27706 over here. 
This change also makes sense to me, but I'd like @jpountz or @jimczi  to also comment on this because fixing it might change the behaviour in unexpected ways or there might be other side effects to consider. 
It looks like we never reached the TextFieldType-case in the past and I'm wondering under which circumstance we might have wanted to.

Reply:
Ping, any chance someone would be able to take a look at this one?